From 0f8abfbabbaf72de81efe477bfedd2c33b06e00f Mon Sep 17 00:00:00 2001
From: Andreas Cord-Landwehr <cordlandwehr@kde.org>
Date: Sun, 12 Feb 2023 09:01:58 +0100
Subject: [PATCH] Use VisionFive2 specific MESA Fork

---
 src/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index f36c238..27df18d 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -51,7 +51,7 @@ target_link_libraries(KPipeWire PRIVATE Qt::Quick Qt::GuiPrivate KF5::I18n
 if (QT_MAJOR_VERSION EQUAL "5")
     target_link_libraries(KPipeWire PRIVATE Qt5::Gui_EGL)
     if (${Qt5Gui_OPENGL_IMPLEMENTATION} STREQUAL "GLESv2")
-        target_link_libraries(KPipeWire PRIVATE Qt5::Gui_GLESv2)
+        target_link_libraries(KPipeWire PRIVATE Qt5::Gui_GLESv2_PVR_MESA)
     else()
         target_link_libraries(KPipeWire PRIVATE Qt5::Gui_GL)
     endif()
-- 
2.37.2

