From 832f0107869472b3335e66250e85446d1767928f Mon Sep 17 00:00:00 2001
From: Andreas Cord-Landwehr <cordlandwehr@kde.org>
Date: Sun, 3 Jan 2021 11:22:47 +0100
Subject: [PATCH] Utilize bitbake download and use tools from PATH

---
 data/CMakeLists.txt | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/data/CMakeLists.txt b/data/CMakeLists.txt
index d3df1d0..aec0718 100644
--- a/data/CMakeLists.txt
+++ b/data/CMakeLists.txt
@@ -20,17 +20,17 @@ set(
 
 set(
     gen_binary_files_BIN
-    ${CMAKE_BINARY_DIR}/utils/storage/gen_binary_files
+    gen_binary_files
 )
 
 set(
     import_interpolation_BIN
-    ${CMAKE_BINARY_DIR}/utils/storage/import_interpolation
+    import_interpolation
 )
 
 set(
     gen_unigram_BIN
-    ${CMAKE_BINARY_DIR}/utils/training/gen_unigram
+    gen_unigram
 )
 
 add_custom_target(
-- 
2.27.0

