diff --git a/kcmkwin/kwindecoration/declarative-plugin/previewitem.cpp b/kcmkwin/kwindecoration/declarative-plugin/previewitem.cpp
index a77261a54..6404ba006 100644
--- a/kcmkwin/kwindecoration/declarative-plugin/previewitem.cpp
+++ b/kcmkwin/kwindecoration/declarative-plugin/previewitem.cpp
@@ -25,6 +25,7 @@
 #include <KDecoration2/DecorationSettings>
 #include <KDecoration2/DecorationShadow>
 #include <KDecoration2/DecoratedClient>
+#include <QCoreApplication>
 #include <QCursor>
 #include <QPainter>
 #include <QQmlContext>
diff --git a/plugins/platforms/x11/standalone/x11_platform.h b/plugins/platforms/x11/standalone/x11_platform.h
index 02a4ae15d..4fd77c6ca 100644
--- a/plugins/platforms/x11/standalone/x11_platform.h
+++ b/plugins/platforms/x11/standalone/x11_platform.h
@@ -24,6 +24,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include <kwin_export.h>
 
 #include <QObject>
+#include <QX11Info>
 
 namespace KWin
 {
diff --git a/plugins/platforms/x11/windowed/x11windowed_backend.cpp b/plugins/platforms/x11/windowed/x11windowed_backend.cpp
index 302bac7f4..27b70810d 100644
--- a/plugins/platforms/x11/windowed/x11windowed_backend.cpp
+++ b/plugins/platforms/x11/windowed/x11windowed_backend.cpp
@@ -40,6 +40,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 // system
 #include <linux/input.h>
 #include <X11/Xlib-xcb.h>
+#include <X11/keysym.h>
 
 namespace KWin
 {
diff --git a/wayland_server.cpp b/wayland_server.cpp
index dd4727727..a64bd10d5 100644
--- a/wayland_server.cpp
+++ b/wayland_server.cpp
@@ -33,7 +33,9 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include <KWayland/Client/surface.h>
 // Server
 #include <KWayland/Server/compositor_interface.h>
+#include <KWayland/Server/datadevice_interface.h>
 #include <KWayland/Server/datadevicemanager_interface.h>
+#include <KWayland/Server/datasource_interface.h>
 #include <KWayland/Server/display.h>
 #include <KWayland/Server/dpms_interface.h>
 #include <KWayland/Server/idle_interface.h>
