diff --git a/kcmkwin/kwindecoration/declarative-plugin/previewitem.cpp b/kcmkwin/kwindecoration/declarative-plugin/previewitem.cpp
index a77261a54..6404ba006 100644
--- a/kcmkwin/kwindecoration/declarative-plugin/previewitem.cpp
+++ b/kcmkwin/kwindecoration/declarative-plugin/previewitem.cpp
@@ -25,6 +25,7 @@
 #include <KDecoration2/DecorationSettings>
 #include <KDecoration2/DecorationShadow>
 #include <KDecoration2/DecoratedClient>
+#include <QCoreApplication>
 #include <QCursor>
 #include <QPainter>
 #include <QQmlContext>
diff --git a/plugins/platforms/x11/common/eglonxbackend.cpp b/plugins/platforms/x11/common/eglonxbackend.cpp
index 1990fc49b..ac3376bfa 100644
--- a/plugins/platforms/x11/common/eglonxbackend.cpp
+++ b/plugins/platforms/x11/common/eglonxbackend.cpp
@@ -272,7 +272,7 @@ EGLSurface EglOnXBackend::createSurface(xcb_window_t window)
         //       cannot be an xcb_window_t.
         surface = eglCreatePlatformWindowSurfaceEXT(eglDisplay(), config(), (void *) &window, nullptr);
     } else {
-        surface = eglCreateWindowSurface(eglDisplay(), config(), window, nullptr);
+        surface = eglCreateWindowSurface(eglDisplay(), config(), (void *) &window, nullptr);
     }
 
     return surface;
diff --git a/plugins/platforms/x11/standalone/x11_platform.h b/plugins/platforms/x11/standalone/x11_platform.h
index 07d79b889..c8b3078af 100644
--- a/plugins/platforms/x11/standalone/x11_platform.h
+++ b/plugins/platforms/x11/standalone/x11_platform.h
@@ -24,6 +24,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include <kwin_export.h>
 
 #include <QObject>
+#include <QX11Info>
 
 #include <memory>
 
diff --git a/plugins/platforms/x11/windowed/x11windowed_backend.cpp b/plugins/platforms/x11/windowed/x11windowed_backend.cpp
index 95306b027..8b7513f68 100644
--- a/plugins/platforms/x11/windowed/x11windowed_backend.cpp
+++ b/plugins/platforms/x11/windowed/x11windowed_backend.cpp
@@ -42,6 +42,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 // system
 #include <linux/input.h>
 #include <X11/Xlib-xcb.h>
+#include <X11/keysym.h>
 
 namespace KWin
 {
