diff --git a/src/tools/CMakeLists.txt b/src/tools/CMakeLists.txt
index 5eea6e49..b78afb7b 100644
--- a/src/tools/CMakeLists.txt
+++ b/src/tools/CMakeLists.txt
@@ -1,3 +1,24 @@
+cmake_minimum_required(VERSION 3.16)
+set(QT_MIN_VERSION "5.15.0")
+project(QtWaylandScannerKde)
+set(CMAKE_C_STANDARD 99)
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+# ECM setup
+set(KDE_COMPILERSETTINGS_LEVEL "5.82")
+include(FeatureSummary)
+find_package(ECM 5.82 NO_MODULE REQUIRED)
+set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH})
+
+include(CheckIncludeFile)
+
+include(KDEInstallDirs)
+include(KDECompilerSettings NO_POLICY_SCOPE)
+include(KDECMakeSettings)
+
+find_package(Qt5 ${QT_MIN_VERSION} CONFIG REQUIRED Concurrent)
+
 remove_definitions(-DQT_NO_CAST_FROM_ASCII)
 add_executable(qtwaylandscanner_kde qtwaylandscanner.cpp)
 target_link_libraries(qtwaylandscanner_kde Qt::Core)
