From dd3565bc3ba01ad6c41425bb0fd831ac79ae2a11 Mon Sep 17 00:00:00 2001
From: Andreas Cord-Landwehr <cordlandwehr@kde.org>
Date: Wed, 8 Feb 2023 20:36:55 +0100
Subject: [PATCH] Use VisionFive2 specific MESA Fork

---
 libtaskmanager/declarative/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libtaskmanager/declarative/CMakeLists.txt b/libtaskmanager/declarative/CMakeLists.txt
index 062efb054..66c70b6ae 100644
--- a/libtaskmanager/declarative/CMakeLists.txt
+++ b/libtaskmanager/declarative/CMakeLists.txt
@@ -37,7 +37,7 @@ target_link_libraries(taskmanagerplugin Qt::Qml Qt::GuiPrivate
                                         Wayland::Client)
 if (QT_MAJOR_VERSION EQUAL "5")
     if (${Qt5Gui_OPENGL_IMPLEMENTATION} STREQUAL "GLESv2")
-        target_link_libraries(taskmanagerplugin Qt5::Gui_GLESv2)
+        target_link_libraries(taskmanagerplugin Qt5::Gui_GLESv2_PVR_MESA)
     elseif(TARGET Qt5::Gui_GL)
         target_link_libraries(taskmanagerplugin Qt5::Gui_GL)
     endif()
-- 
2.37.2

